lib/unifex/code_generator/base_types/list.ex

```diff
@@ -11,6 +11,25 @@ defmodule Unifex.CodeGenerator.BaseTypes.List do
   alias Unifex.CodeGenerator.BaseType
 
   @impl BaseType
+  def generate_native_type(ctx)
+
+  def generate_native_type(%{mode: :const_if_not_ptr_on_ptr} = ctx) do
+    ptr? =
+      BaseType.generate_native_type(ctx.subtype, ctx.mode, ctx.generator, %{ctx | mode: :default})
+      |> String.replace(" ", "")
+      |> String.match?(~r"\*$")
```

Comment:
Maybe we should have a `ptr_level(ctx)` callback, returning the number of `*` in the type, returning 0 by default? Then we could just call it in `BaseType.generate_native_type` and check if it's `>= 2`